File: Aula_04/listapoo01_ex03.java
class Conta {
    public String titular;                // Declaração dos atributos
    public String numero;
    public double saldo;
    public Conta() {                      // Construtor
        titular = "sem nome";             // Atributos
        numero = "sem número";
        saldo = 0;
    }
    public void depositar(double valor) {
        saldo += valor;
    }    
    public void sacar(double valor) {
        saldo -= valor;
    }
}

public class listapoo01_ex03 {
    public static void main(String[] args) {
        Conta x = new Conta();    // x é uma referência
        System.out.println(x.titular + " " + x.numero + " " + x.saldo);
        x.titular = "Raquel";
        x.numero = "123-x";
        x.saldo = 1000;
        System.out.println(x.titular + " " + x.numero + " " + x.saldo);
        x.depositar(500);
        System.out.println(x.titular + " " + x.numero + " " + x.saldo);
    }
}

File: Aula_04/listapoo01_ex03.cpp
#include <iostream>

using namespace std;

class Conta {
    public:
    string titular;  // Declaração dos atributos
    string numero;
    double saldo;
    Conta() {                      // Construtor
        titular = "sem nome";      // Atributos
        numero = "sem número";
        saldo = 0;
    }
    void depositar(double valor) {
        saldo += valor;
    }    
    void sacar(double valor) {
        saldo -= valor;
    }
};

int main() {
    Conta x;                // x é uma instância
    Conta& y = x;           // y é uma referência para x
    Conta* z = new Conta(); // z é um ponteiro para a instância
    cout << x.titular << " " << x.numero << " " << x.saldo << endl;
    x.titular = "Raquel";
    x.numero = "123-x";
    x.saldo = 1000;
    cout << x.titular << " " << x.numero << " " << x.saldo << endl;
    x.depositar(500);
    cout << x.titular << " " << x.numero << " " << x.saldo << endl;
    y.depositar(1000);
    cout << x.titular << " " << x.numero << " " << x.saldo << endl;
    z->numero = "123";
    (*z).numero = "456";
    delete z;
    return 0;
}


